website/addons/figshare/views/config.py

```diff
@@ -38,16 +38,31 @@ def figshare_config_put(node_addon, auth, **kwargs):
     fields = request.json.get('selected', {})
     node = node_addon.owner
 
-    folder = {
-        'name': fields.get('name') or '',
-        'id': fields.get('id') or None,
-        'type': fields.get('type') or None,
-    }
-    node_addon.update_fields(folder, node, auth)
+    name = fields.get('name')
+    figshare_id = fields.get('id')
+    figshare_type = fields.get('type')
+
+    if not name or not figshare_id or not figshare_type:
+        return HTTPError(http.BAD_REQUEST, message="""
```

Comment:
> Use parentheses for line continuations.

https://cosdev.readthedocs.org/en/latest/style_guides/python.html#line-lengths

Also, I'm not sure how these line breaks and indents will display in a JSON response. Maybe we go with a shorter message, like: "Must provide name, id, and type".